### PR TITLE
Migrate AgentResult and Permissions to S7 values

### LINKED ISSUE #117: Migrate AgentResult and Permissions to S7 value contracts
## Direction

Follow #59 and ADR-0009: Deputy value contracts use S7; mutable runtime owners retain R6. The HookMatcher/AgentEvent migration establishes read-only stored properties, released S7 registration, and composition with ellmer. This follow-up covers the two classes explicitly identified in #59's audit comment.

## AgentResult

Migrate the run result to a Deputy-owned S7 class containing original ellmer turns and Deputy events. Decide and document which currently writable fields become read-only; preserve result inspection helpers, original conditions/provider content, usage, and canonical run context. A result spans a governed run, so it must not subclass ellmer Turn or Round. Update constructors, consumers, examples, exports, and serialization tests together.

## Permissions

Migrate configuration to an S7 value while keeping policy evaluation explicit. Preserve canonical grants, immutable authority ceilings, mode narrowing, callback behavior, and fail-closed evaluation. Test attempted widening through delegated runs and configuration replacement, including nullable properties and serialization. The host callback retains caller-owned executable state; do not claim deep immutability or serialize authority into a portable grant.

## Acceptance

- Both contracts use S7, with no parallel S3 value API or R6 constructor facades.
- Public API changes and ownership semantics are documented with runnable examples.
- Existing result inspection and permission/delegation regression coverage passes; add focused tests for the new value and registration boundaries.
- Record fresh-process and installed-package validation against released ellmer/S7, plus full package checks and PR review.

Remaining usage, limits, definitions, context policy, hook/permission result records, and Skill should migrate in subsequent bounded slices under the same S7 direction. Agent, LeadAgent, HookRegistry, and FileCheckpointStore keep reference identity.

### PR DESCRIPTION
**Pull request by JamesHWade**
Closes #117.

`AgentResult(...)` and `Permissions(...)` now create Deputy-owned S7 values with read-only properties. Completed results preserve original turns, events, conditions, usage, and canonical run context; all previously writable result fields are now read-only. Policy evaluation moves to `permissions_check()` with explicit internal functions, retaining gating precedence, callback behavior, canonical grants, mode narrowing, and delegated authority ceilings. Permission constructors reject malformed capability flags and callbacks, and Agent construction requires an S7 Permissions value.

This is an intentional API change: replace `$new()` constructors with `AgentResult(...)` and `Permissions(...)`; replace result instance methods with `result_n_turns()`, `result_tool_calls()`, `result_tool_results()`, `result_text_chunks()`, and `result_is_success()`. Replace `policy$check(...)` with `permissions_check(policy, ...)`. `$` property reads remain available. Examples, consumers, generated reference docs, README, and namespace registration are updated together.

```r
library(deputy)
policy <- Permissions(file_write = FALSE)
permissions_check(policy, "write_file", list(path = "output.txt"))
result <- AgentResult(response = "Done", events = list(AgentEvent("text", text = "Done")))
result_is_success(result)
result_text_chunks(result)
```

[ADR-0010](https://github.com/JamesHWade/deputy/blob/codex/s7-results-permissions/dev/adr/0010-s7-results-permissions.md) records the property contracts, ellmer composition, policy ownership, and serialization limits. Read-only records do not deep-copy embedded environments or callbacks, and serialized policies do not create portable grants or replace an active Agent's authority.

Remaining value families are tracked in #119.

Validation at `379aa7d`: the full local suite passes 4,228 assertions with no failures or errors (two expected filesystem warnings from the existing write-error test and six skips). Installed-package R CMD check reports 0 errors, 0 warnings, and 0 notes. Targeted result, permission, delegation, printing, and fresh-process serialization tests pass, as do Air, Jarl, pkgdown reference checks, and the full site build. The final revision passes CI on macOS, Windows, and Ubuntu, along with the package executable, pkgdown, formatting, Claude review, test coverage, and both Codecov checks. Validation uses released ellmer 0.5.0 and S7 0.2.2; the declared R 4.1 floor was not directly tested. Roxygen2 8.0.0 generated documentation locally while the repository records 8.1.0.

Review follow-up: missing/non-character text payloads are ignored, character-vector payloads are flattened in event order, and printed policies distinguish NULL from empty tool gates. All three review threads are addressed and resolved.

### CODE REVIEW

File: R/agent-result.R [hunk 1]
```diff
@@ -165,221 +165,308 @@ S7::method(print, AgentEvent) <- function(x, ...) {
...
-    #' @description
-    #' Check if the agent completed successfully.
-    #' @return Logical indicating success
-    is_success = function() {
-      self$stop_reason == "complete"
-    },
+S7::method(result_text_chunks, AgentResult) <- function(result) {
+  text_events <- Filter(function(e) e$type == "text", result@events)
+  vapply(text_events, function(e) e$text, character(1))
+}
```
[the remark this answers is not in the document]
Comment (JamesHWade) [reply R/agent-result.R:417]: Fixed in 379aa7d. Text inspection ignores missing and non-character payloads, flattens character vectors in event order, and returns character(0) when no text remains. The regression covers absent, NULL, empty, numeric, list, scalar, and vector payloads.

File: R/permissions.R [hunk 2]
```diff
@@ -1,846 +1,391 @@
...
     }
-  )
-)
+    cli::cli_text("tool_allowlist: {tool_allowlist}")
+    tool_denylist <- if (length(x@tool_denylist) == 0L) {
+      "NULL"
+    } else {
+      paste(x@tool_denylist, collapse = ", ")
+    }
+    cli::cli_text("tool_denylist: {tool_denylist}")
```
[the remark this answers is not in the document]
Comment (JamesHWade) [reply R/permissions.R]: Fixed in 379aa7d. Policy summaries now print NULL and character(0) distinctly for both tool gates. The regression verifies the output and confirms that a known read tool is allowed with no allowlist but denied by an empty allowlist.

File: R/agent-result.R [hunk 3]
```diff
@@ -165,221 +165,308 @@ S7::method(print, AgentEvent) <- function(x, ...) {
...
 
-    #' @description
-    #' Check if the agent completed successfully.
-    #' @return Logical indicating success
-    is_success = function() {
-      self$stop_reason == "complete"
-    },
+S7::method(result_text_chunks, AgentResult) <- function(result) {
+  text_events <- Filter(function(e) e$type == "text", result@events)
+  vapply(text_events, function(e) e$text, character(1))
```
[the remark this answers is not in the document]
Comment (JamesHWade) [reply R/agent-result.R]: Fixed in 379aa7d. The extractor ignores missing/non-character payloads and flattens character vectors in event order. Its return remains a character vector, including the empty case. The documented behavior and regression tests cover constructible missing, empty, non-character, and non-scalar payloads.
